api/api-shared-v6/src/main/java/com/thoughtworks/go/apiv6/shared/representers/stages/JobRepresenter.java

```diff
@@ -0,0 +1,229 @@
+        }
+    }
+
+    private static void addRunInstanceCount(OutputWriter outputWriter, JobConfig jobConfig) {
+        if (jobConfig.isRunOnAllAgents()) {
+            outputWriter.add("run_instance_count", "all");
+        } else if (jobConfig.getRunInstanceCount() != null && !jobConfig.getRunInstanceCount().isEmpty()) {
+            outputWriter.add("run_instance_count", jobConfig.getRunInstanceCountValue());
+        } else {
+            outputWriter.add("run_instance_count", (String) null);
+        }
+    }
+
+    public static JobConfig fromJSON(JsonReader jsonReader) {
+        JobRepresenter.jsonReader = jsonReader;
+        JobConfig jobConfig = new JobConfig();
+        if (jsonReader == null) {
+            return jobConfig;
+        }
+        jsonReader.readCaseInsensitiveStringIfPresent("name", jobConfig::setName);
+        setRunInstanceCount(jobConfig);
+        setTimeout(jobConfig);
+        jsonReader.readStringIfPresent("elastic_profile_id", jobConfig::setElasticProfileId);
+        setArtifacts(jobConfig);
+        setEnvironmentVariables(jobConfig);
+        setResources(jobConfig);
+        setTabs(jobConfig);
+        setProperties(jobConfig);
+        setTasks(jobConfig);
+
+        return jobConfig;
+    }
+
+    private static void setTasks(JobConfig jobConfig) {
+        Tasks allTasks = new Tasks();
+        jsonReader.readArrayIfPresent("tasks", tasks -> {
+            tasks.forEach(task -> {
+                allTasks.add(TaskRepresenter.fromJSON(new JsonReader(task.getAsJsonObject())));
+            });
+        });
+
+        jobConfig.setTasks(allTasks);
+    }
+
+    private static void setProperties(JobConfig jobConfig) {
+        ArtifactPropertiesConfig artifactPropertyConfigs = new ArtifactPropertiesConfig();
+        jsonReader.readArrayIfPresent("properties", properties -> {
+            properties.forEach(property -> {
+                artifactPropertyConfigs.add(PropertyConfigRepresenter.fromJSON(new JsonReader(property.getAsJsonObject())));
+            });
+        });
+
+        jobConfig.setProperties(artifactPropertyConfigs);
+    }
+
+    private static void setTabs(JobConfig jobConfig) {
+        Tabs tabsConfig = new Tabs();
+        jsonReader.readArrayIfPresent("tabs", tabs -> {
+            tabs.forEach(tab -> {
+                tabsConfig.add(TabConfigRepresenter.fromJSON(new JsonReader(tab.getAsJsonObject())));
+            });
+        });
+        jobConfig.setTabs(tabsConfig);
+    }
+
+    private static void setResources(JobConfig jobConfig) {
+        ResourceConfigs resourceConfigs = new ResourceConfigs();
+        jsonReader.readArrayIfPresent("resources", resources -> {
+            resources.forEach(resource -> {
+                resourceConfigs.add(new ResourceConfig(resource.getAsString()));
+            });
+        });
+
+        jobConfig.setResourceConfigs(resourceConfigs);
+    }
+
+    private static void setEnvironmentVariables(JobConfig jobConfig) {
```

Comment:
This method seems like a duplicate in 3 places. Also see `EnvrionmentVariableRepresenter`